### PR TITLE
Do not convert Chinese Traditional to Simplified for the target language

### LINKED ISSUE #1033: Improve translation quality of English to Chinese
The quality of the teacher model doesn't look great: COMET 85.67 vs Google Translate 89.45 (-4.41%) on flores-test. 

I suspect we have some issues in the training data:
1. I found that we used dataset `mtdata_Statmt-backtrans_enzh-wmt20-eng-zho` which is backtranslations for zh-en WMT20 task. So the zh part of it might be of poor quality. We already have our own back-translations.
2. We convert all the data from Traditional to Simplified with hanzi-convert. People on the internet say such conversion is imprecise due to the regional differences, vocabulary, word composition etc., so it is not equal to a proper translation. It might be the case that it pollutes data when Chinese is the target language. We can just filter all the data in Traditional and see if it makes a difference.
3. We use a 64k joint vocabulary for very different scripts, it's worth running an experiment with separate vocabularies 32k in size. Depends on https://github.com/mozilla/translations/issues/913
4. There are known issues for some datasets: [UN](https://github.com/mozilla/translations/issues/744), [HPLT](https://github.com/mozilla/translations/issues/1030), [desegmentation](https://github.com/mozilla/translations/issues/743). We can address them with ad-hoc fixes through OpusCleaner and the mono-cleaning scripts.

The question is which of those issues contribute most. It would require running separate experiments for most of them to see if it moves the needle.

Another indicator that something is wrong with the data is that one of my experiments completely diverged and I had to reduce the learning rate. Also when training in two stages there is a big bump in the cost when moving to training on the original parallel corpus only. This typically indicates that the fine-tuning data is noisier compared to the pre-training one (a mix of back-translated and original corpus).

Image

### PR DESCRIPTION
**Pull request by eu9ene**
closes #1033

### CODE REVIEW

File: pipeline/data/download-mono.py [hunk 1]
```diff
@@ -133,20 +133,37 @@ def main(args_list: Optional[list[str]] = None) -> None:
...
-        converted_path = file_destination.with_suffix(".converted.zst")
-        stats = chinese_converter.convert_file(
-            file_destination, converted_path, ChineseType.simplified
-        )
-        shutil.move(converted_path, file_destination)
-        print(
-            f"Converted {stats.script_conversion.converted} lines from {stats.script_conversion.visited} to Chinese Simplified"
-        )
-        stats.save_json()
+        if os.environ["SRC"] == "zh":
```
Comment (gregtatum) [inline pipeline/data/download-mono.py]: Can we pass this in as an argument in argparse rather than rely on the environment variables? I'd like to not rely on two API surfaces for our python scripts, as it's confusing on what the inputs are.

File: pipeline/data/download-mono.py [hunk 2]
```diff
@@ -133,20 +133,37 @@ def main(args_list: Optional[list[str]] = None) -> None:
...
+            logger.info("Converting the output file to Chinese Simplified")
+            chinese_converter = ChineseConverter()
+            converted_path = file_destination.with_suffix(".converted.zst")
+            stats = chinese_converter.convert_file(
+                file_destination, converted_path, ChineseType.simplified
+            )
+            shutil.move(converted_path, file_destination)
```
Comment (gregtatum) [inline pipeline/data/download-mono.py]: I guess this move overwrites the file_destination file, so we don't retain extra copies of the datasets. It might be more explicit to remove the old one and then move so it's clearer. I had to look up this API.

File: pipeline/data/cjk.py [hunk 3]
```diff
@@ -102,3 +134,50 @@ def _convert_line(self, text: str, to: ChineseType) -> str:
...
+def handle_chinese_mono(file_destination: Path, is_src: bool, variant: ChineseType):
+    converted_path = file_destination.with_suffix(".converted.zst")
+    chinese_converter = ChineseConverter()
+    if is_src:
+        logger.info(f"Converting the output file to {variant}")
+        stats = chinese_converter.convert_file(file_destination, converted_path, variant)
+    else:
+        logger.info(f"Filtering out everything except {variant} in the output file")
+        stats = chinese_converter.filter_file(file_destination, converted_path, variant)
+    shutil.move(converted_path, file_destination)
```
Comment (gregtatum) [inline pipeline/data/cjk.py:148]: Stylistically, since we're using `pathlib`, I'd prefer to use its methods like `.rename` and `.unlink` rather than using multiple libraries like `shutil`, unless there is a good reason to do so. I generally prefer `pathlib` over other file and file path manipulation tools, as it's pretty ergonomic to work with.

**Review by gregtatum — APPROVED** [review]

File: pipeline/data/cjk.py (hunk 3 above)
Comment (eu9ene) [reply pipeline/data/cjk.py]: `.replace` similar to `.move` does it in one line

**Review by bhearsum — APPROVED** [review]
